From f48448bd4006f2e6f64736128cd5b5ba9e49c45b Mon Sep 17 00:00:00 2001
From: Sanjay R Mehta <Sanju.Mehta@amd.com>
Date: Mon, 18 Dec 2017 20:48:30 +0530
Subject: [PATCH] ASoC: AMD: Enable pci bus-mastering for ACP3.x device

By default DMA bus-mastering is disabled in uefi boot
mode for ACP3.x device, so pci_set_master() api need
to be invoked explicitly to set the DMA bit in the
ACP3.x PCI_COMMAND register.

Signed-off-by: Sanjay R Mehta <Sanju.Mehta@amd.com>
Signed-off-by: Vijendar Mukunda <Vijendar.Mukunda@amd.com>
---
 sound/soc/amd/raven/pci-acp3x.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sound/soc/amd/raven/pci-acp3x.c b/sound/soc/amd/raven/pci-acp3x.c
index 5891c5b..0d39aa8 100644
--- a/sound/soc/amd/raven/pci-acp3x.c
+++ b/sound/soc/amd/raven/pci-acp3x.c
@@ -70,6 +70,9 @@ static int snd_acp3x_probe(struct pci_dev *pci,
 		goto release_regions;
 	}
 
+	/* set pci bus-mastering */
+	pci_set_master(pci);
+
 	pci_set_drvdata(pci, adata);
 
 	val = rv_readl(adata->acp3x_base + mmACP_I2S_PIN_CONFIG);
-- 
2.7.4

